src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -295,6 +295,7 @@ ucs_status_t uct_dc_mlx5_ep_am_short(uct_ep_h tl_ep, uint8_t id, uint64_t hdr,
 #if HAVE_IBV_EXP_DM
     }
 
+    UCT_CHECK_AM_ID(id);
```

Comment:
fixing ud/dc should be in another PR

Reply:
#3234 has been created to fix this comment